detekt-rules-coroutines/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/coroutines/SuspendFunSwallowedCancellation.kt

```diff
@@ -0,0 +1,180 @@
+            "`runCatching` catches all the `Exception`s.",
+        debt = Debt.TEN_MINS
+    )
+
+    override fun visitCallExpression(expression: KtCallExpression) {
+        super.visitCallExpression(expression)
+
+        val resultingDescriptor = expression.getResolvedCall(bindingContext)?.resultingDescriptor
+        resultingDescriptor ?: return
+        if (resultingDescriptor.fqNameSafe != RUN_CATCHING_FQ) return
+
+        expression.forEachDescendantOfType<KtCallExpression>({
+            expression == it || shouldTraverseInside(it, bindingContext)
+        }) { descendant ->
+            val callableDescriptor = descendant.getResolvedCall(bindingContext)?.resultingDescriptor
+            if (callableDescriptor?.isSuspend == true) {
+                report(
+                    message = "The suspend function call ${callableDescriptor.fqNameSafe.shortName()} called inside " +
+                        "`runCatching`. You should either use specific `try-catch` only catching exception that you " +
+                        "are expecting or rethrow the `CancellationException` if already caught.",
+                    expression
+                )
+            }
+        }
+
+        expression.forEachDescendantOfType<KtForExpression> { descendant ->
+            if (descendant.doesLoopHasSuspendingIterators()) {
+                report(
+                    message = "The for-loop expression has suspending operator which is called inside " +
+                        "`runCatching`. You should either use specific `try-catch` only catching exception that you " +
+                        "are expecting or rethrow the `CancellationException` if already caught.",
+                    expression
+                )
+            }
+        }
+    }
+
+    @Suppress("ReturnCount")
+    private fun shouldTraverseInside(psiElement: PsiElement, bindingContext: BindingContext): Boolean {
+        if (psiElement is KtCallExpression) {
+            val callableDescriptor =
+                psiElement.getResolvedCall(bindingContext)?.resultingDescriptor as? FunctionDescriptor
+            callableDescriptor ?: return false
+
+            if (callableDescriptor.fqNameSafe == RUN_CATCHING_FQ) return false
+
+            if (callableDescriptor.isInline.not()) return false
+        }
+
+        if (psiElement is KtValueArgument) {
+            val callExpression = psiElement.getParentOfType<KtCallExpression>(true)
+            callExpression ?: return false
+            val valueParameterDescriptor =
+                callExpression.getResolvedCall(bindingContext)?.getParameterForArgument(psiElement)
+            valueParameterDescriptor ?: return false
+            if (valueParameterDescriptor.isCrossinline || valueParameterDescriptor.isNoinline) return false
+        }
+        return true
+    }
+
+    private fun KtForExpression.doesLoopHasSuspendingIterators(): Boolean {
```

Comment:
"loop" is contained in receiver type
```suggestion
    private fun KtForExpression.hasSuspendingIterators(): Boolean {
```
or
```suggestion
    private fun KtForExpression.hasSuspendingIteratorsInLoop(): Boolean {
```